[CPP20] maybe-uninitialized warning in MeasurementTrackerImpl::addPhase2Det

```
In member function '__ct ',
    inlined from 'construct_at' at /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/include/c++/12.3.1/bits/stl_construct.h:97:14,
    inlined from 'construct' at /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/include/c++/12.3.1/bits/alloc_traits.h:518:21,
    inlined from '_M_realloc_insert' at /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/include/c++/12.3.1/bits/vector.tcc:462:28,
    inlined from 'emplace_back' at /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/include/c++/12.3.1/bits/vector.tcc:123:21,
    inlined from 'push_back' at /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/include/c++/12.3.1/bits/stl_vector.h:1294:21,
    inlined from 'addPhase2Det' at /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/bde6290197b40a7abece5e0ce6f36880/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_13_3_CPP20_X_2023-08-21-1100/src/RecoTracker/MeasurementDet/plugins/MeasurementTrackerImpl.cc:286:28,
    inlined from 'addDets' at /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/bde6290197b40a7abece5e0ce6f36880/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_13_3_CPP20_X_2023-08-21-1100/src/RecoTracker/MeasurementDet/plugins/MeasurementTrackerImpl.cc:236:23:
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/bde6290197b40a7abece5e0ce6f36880/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_13_3_CPP20_X_2023-08-21-1100/src/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.h:16:18: warning: 'MEM <unsigned long> [(unsigned int *)&D.14013 + 48B]' may be used uninitialized [-Wmaybe-uninitialized]
    16 | class dso_hidden TkPhase2OTMeasurementDet final : public MeasurementDet {
      |                  ^
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/bde6290197b40a7abece5e0ce6f36880/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_13_3_CPP20_X_2023-08-21-1100/src/RecoTracker/MeasurementDet/plugins/MeasurementTrackerImpl.cc: In member function 'addDets':
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/bde6290197b40a7abece5e0ce6f36880/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_13_3_CPP20_X_2023-08-21-1100/src/RecoTracker/MeasurementDet/plugins/MeasurementTrackerImpl.cc:286:80: note: '<anonymous>' declared here
  286 |     thePhase2Dets.push_back(TkPhase2OTMeasurementDet(gd, thePhase2DetConditions));
      |                                                                                ^
Leaving library rule at src/RecoTracker/MeasurementDet/plugins
```

Full log: https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc12/CMSSW_13_3_CPP20_X_2023-08-21-1100/RecoTracker/MeasurementDet

--- a/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.cc
+++ b/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.cc
@@ -8,7 +8,7 @@
 #include "TrackingTools/PatternTools/interface/TrajMeasLessEstim.h"
 
 TkPhase2OTMeasurementDet::TkPhase2OTMeasurementDet(const GeomDet* gdet, Phase2OTMeasurementConditionSet& conditions)
-    : MeasurementDet(gdet), theDetConditions(&conditions) {
+    : MeasurementDet(gdet), theDetConditions(&conditions), index_(0) {
   if (dynamic_cast<const PixelGeomDetUnit*>(gdet) == nullptr) {
     throw MeasurementDetException(
         "TkPhase2OTMeasurementDet constructed with a GeomDet which is not a PixelGeomDetUnit");

--- a/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.cc
+++ b/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.cc
@@ -8,7 +8,7 @@
 #include "TrackingTools/PatternTools/interface/TrajMeasLessEstim.h"
 
 TkPhase2OTMeasurementDet::TkPhase2OTMeasurementDet(const GeomDet* gdet, Phase2OTMeasurementConditionSet& conditions)
-    : MeasurementDet(gdet), theDetConditions(&conditions), index_(0) {
+    : MeasurementDet(gdet), index_(0), theDetConditions(&conditions) {
   if (dynamic_cast<const PixelGeomDetUnit*>(gdet) == nullptr) {
     throw MeasurementDetException(
         "TkPhase2OTMeasurementDet constructed with a GeomDet which is not a PixelGeomDetUnit");

--- a/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.h
+++ b/RecoTracker/MeasurementDet/plugins/TkPhase2OTMeasurementDet.h
@@ -80,7 +80,6 @@ public:
   void setIndex(int i) { index_ = i; }
 
 private:
-  unsigned int id_;
   //  std::vector< LocalPoint > badRocPositions_;
 
   int index_;

